### ISSUE TITLE
[maestro++] Minimize old Auto-Merge Status comment as outdated when posting a new one

### ISSUE THREAD

**Issue by dagood** [dotnet/arcade#4346]
The auto-merge status comments are pretty noisy when trying to read a conversation in an auto-PR, for example https://github.com/dotnet/corefx/pull/42511:

> ![image](https://user-images.githubusercontent.com/12819531/68788622-5f34f580-0609-11ea-9b9b-642682e0ec1e.png)

The semi-new minimizing ("hide") feature makes it easier to follow (https://github.com/dotnet/core-setup/pull/8775):

> ![image](https://user-images.githubusercontent.com/12819531/68787954-12045400-0608-11ea-8946-6e865e9175b1.png)

Not going all the way and deleting the old comments makes sense in case it somehow misfires, or if there's some info in there that's useful to diagnose something. These are pretty hefty comments to have in the middle of conversations though.

Note that the v3 API doesn't support this, I believe the v4 API (graphql) would be required: https://github.com/isaacs/github/issues/1480.

---

This is a nice-to-have, I've sometimes minimized the comments myself when I have the permissions in the repo. I wanted to at least raise the suggestion. 😄

**Comment by riarenas** [comment]
Definitely nice to have and it's not the first time something to stop the comment spamming every 5 minutes has been requested. We were thinking of conveying this information using the checks in the PR instead of adding a comment every time as part of https://github.com/dotnet/arcade/issues/3326.

Wonder if this would be a quicker way to get some relief in that area while we can invest to moving to the checks API? @mmitche

**Comment by dagood** [comment]
Ah yeah, forgot about that one.

Deleting seems easier if it's ok since it's supported in v3, using v4 for this in particular just to add hiding might not be so easy depending on what the Maestro++ code's like here. (There's also the issue of how Maestro++ is iterating through past comments... if it's terrible to check more than only the last one, or what.)

**Comment by dagood** [comment]
We can get some pain relief by doing what I did with VersionTools and just put the large text in a summary:

> ## Auto-Merge Status
> This pull request has not been merged because Maestro++ is waiting on the following merge policies.
> 
> - ❌ **Standard Merge Policies** Unsuccessful checks: corefx-ci, corefx-ci (MacOS Build x64_Debug), corefx-ci (Linux Build musl_x64_Debug), corefx-ci (Linux Build arm64_Debug), corefx-ci (Linux Build x64_Debug), corefx-ci (Linux Build musl_arm64_Debug), corefx-ci (Linux Build RedHat6_x64_Release), corefx-ci (Linux Build arm_Debug), corefx-ci (Linux Build wasm_Release), corefx-ci (Windows Build NETFX_x86_Release), corefx-ci (Windows Build x86_Release), corefx-ci (Windows Build x64_Debug), corefx-ci (Windows Packaging All Configurations x64_Debug)
> - ✔️ **Standard Merge Policies** Succeeded - No reviews have requested changes.

vs.

> Auto-Merge Status
> <details><summary>This pull request has not been merged because Maestro++ is waiting on  merge policies. (Click me for details.)</summary>
> 
> - ❌ **Standard Merge Policies** Unsuccessful checks: corefx-ci, corefx-ci (MacOS Build x64_Debug), corefx-ci (Linux Build musl_x64_Debug), corefx-ci (Linux Build arm64_Debug), corefx-ci (Linux Build x64_Debug), corefx-ci (Linux Build musl_arm64_Debug), corefx-ci (Linux Build RedHat6_x64_Release), corefx-ci (Linux Build arm_Debug), corefx-ci (Linux Build wasm_Release), corefx-ci (Windows Build NETFX_x86_Release), corefx-ci (Windows Build x86_Release), corefx-ci (Windows Build x64_Debug), corefx-ci (Windows Packaging All Configurations x64_Debug)
> - ✔️ **Standard Merge Policies** Succeeded - No reviews have requested changes.
> 
> </details>

Basically pre-minimize.

**Comment by dagood** [comment]
I have a little C# tool I use for this now: https://gist.github.com/dagood/93ee20f83b558a240e05ea47ef3f4811.

**Comment by markwilkie** [comment]
adding to the dependency epic for triage.

**Comment by mmitche** [comment]
We could do this, or we could also get rid of the comments altogether and use the github checks api to create/update the merge status. That is more ideal. I think there is an issue open for this already.
